orchagent/copporch.cpp

```diff
@@ -869,3 +653,516 @@ void CoppOrch::doTask(Consumer &consumer)
         }
     }
 }
+
+/* Check if the Copp Group name is different while all trap ids remain the same
+ * This is going to be warm boot scenario from an old image which has different naming scheme 
+ */
+bool CoppOrch::checkDupCoppGrpNameAndUpdate(string new_copp_grp_name,
+                                            vector<FieldValueTuple> &fv_tuple)
+{
+    vector<string> trap_id_list;
+    sai_object_id_t trap_group_obj = SAI_NULL_OBJECT_ID;
+    vector<FieldValueTuple> mod_fv_tuple;
+    vector<sai_hostif_trap_type_t> trap_ids;
+    bool ret = false;
+
+    if (m_trap_group_map.find(new_copp_grp_name) != m_trap_group_map.end())
+    {
+        return false;
+    }
+
+    for (auto i = fv_tuple.begin(); i != fv_tuple.end(); i++)
+    {
+        if (fvField(*i) == copp_trap_id_list)
+        {
+            trap_id_list = tokenize(fvValue(*i), list_item_delimiter);
+            getTrapIdList(trap_id_list, trap_ids);
+        }
+    }
+
+    /* In the case where NAT is not supported, there will be no trap IDs associated with
+     * the nat trap group and hence group ID cannot be figured out by iterating through traps.
+     * Using the hardcoded name for this specific case
+     */
+    if (trap_ids.size() == 0)
+    {
+        if (!gIsNatSupported)
```

Comment:
Why the NAT check?

Reply:
This was added earlier for warmboot compatibility. Since now we erase the appDB table during warmboot we can remove it.